src/chain/shared_state.rs

```diff
@@ -493,32 +498,54 @@ mod test {
     //           the index is the index in the `updates` vector, which should have generated the
     //           key we expect to get for the given prefix
     fn update_keys_and_check(start_pfx: &str, updates: Vec<&str>, expected: Vec<(&str, usize)>) {
+        update_keys_and_check_with_version(
```

Comment:
why need this extra function level?

Reply:
We use both, one that does not care about version, and one to test out of order.

Reply:
But in the places of currently calling `update_keys_and_check`, you can also just call `update_keys_and_check_with_version` directly.
It's just an extra covertion into enumerate required when you do so

Reply:
Yes, it automatically generate the version number so the test cases are not polluted with extra information that is not relevant to it.